compiler/src/dotty/tools/backend/sjs/JUnitBootstrappers.scala

```diff
@@ -19,7 +19,84 @@ import Types._
 
 import dotty.tools.dotc.transform.MegaPhase._
 
-/** Generates JUnit bootstrapper classes for Scala.js. */
+/** Generates JUnit bootstrapper objects for Scala.js.
+ *
+ *  On the JVM, JUnit uses run-time reflection to list and invoke JUnit-related
+ *  methods. They are identified by annotations such as `@Test`, `@Before`,
+ *  etc. In Scala.js, there is no such reflection for methods and annotations,
+ *  so a different is used: this phase performs the necessary inspections at
+ *  compile-time, and generates a so-called bootstrapper object where all those
+ *  metadata have been reified.
+ *
+ *  With an example: given the following JUnit test class:
+ *
+ *  ```
+ *  class MyTest {
+ *    @Before def myBefore(): Unit = ...
+ *    @Before def otherBefore(): Unit = ...
+ *
+ *    @Test def myTest1(): Unit = ...
+ *    @Test def myTest2(): Future[Unit] = ...
+ *
+ *    @Ignore @Test def ignoredTest(): Unit = ...
+ *  }
+ *
+ *  object MyTest {
+ *    @AfterClass def myAfterClass(): Unit = ...
+ *  }
+ *  ```
+ *
+ *  this phase generates the following bootstrapper module class:
+ *
+ *  ```
+ *  object MyTest$scalajs$junit$bootstrapper extends Object with Bootstrapper {
+ *    def beforeClass(): Unit = ()
```

Comment:
Why is this empty ?

Reply:
Because there is no `@BeforeClass` method in `object MyTest`.